thread/common/omrthread.c

```diff
@@ -4117,11 +4142,20 @@ monitor_exit(omrthread_t self, omrthread_monitor_t monitor)
 		UPDATE_JLM_MON_EXIT(self, monitor);
 
 #ifdef OMR_THR_THREE_TIER_LOCKING
+#if defined(OMR_THR_SYNCHRONIZATION_PROTOTYPES)
+		omrthread_spinlock_swapState(monitor, J9THREAD_MONITOR_SPINLOCK_UNOWNED);
```

Comment:
There are three spin lock states:

1) UNOWNED - the monitor is unowned

2) OWNED - the monitor is owned

3) EXCEEDED - the monitor is owned and there may be one or more threads waiting for ownership

Currently, we wake up all threads when the spin lock state is EXCEEDED.

Now, we added an optimization -- don't wake threads when threads are spinning. Also, we limited the number of threads that can be woken up.

We can't use the new optimizations and rely upon the EXCEEDED spin lock state at the same since it leads to a deadlock state. Thus, we ignore EXCEEDED.